src/components/recordedit/select-all-row.tsx

```diff
@@ -0,0 +1,405 @@
+
+            // see if the input is empty
+            let temp = !Boolean(data[n]);
+            if (columnModel.column.type.name === 'boolean') {
+                temp = typeof data[n] !== 'boolean';
+            }
+
+            if (isEmpty !== temp) {
+                setIsEmpty(temp);
+            }
+        });
+        return () => subscribe.unsubscribe();
+    }, [watch, isEmpty]);
+
+    // ------------------------ callbacks -----------------------------------//
+    const applyValueToAll = () => {
+        const cm = columnModels[columnModelIndex];
+
+        logRecordeditClientAction(
+            LogActions.SET_ALL_APPLY,
+            cm.logStackPathChild,
+            cm.logStackNode,
+            undefined,
+            cm.column.reference ? cm.column.reference : undefined
+        );
+
+        setValueForAllInputs();
+    };
+
+    const clearAllValues = () => {
+        const cm = columnModels[columnModelIndex];
+
+        logRecordeditClientAction(
+            LogActions.SET_ALL_CLEAR,
+            cm.logStackPathChild,
+            cm.logStackNode,
+            undefined,
+            cm.column.reference ? cm.column.reference : undefined
+        );
+
+        setValueForAllInputs(true);
+    };
+
+    const closeSelectAll = () => {
+        const cm = columnModels[columnModelIndex];
+
+        logRecordeditClientAction(
+            LogActions.SET_ALL_CANCEL,
+            cm.logStackPathChild,
+            cm.logStackNode,
+            undefined,
+            cm.column.reference ? cm.column.reference : undefined
+        );
+
+        toggleActiveSelectAll(columnModelIndex);
+    };
+
+    // Call back for select all checkbox to toggle all forms as selected and unselected
+    const onSelectChange = () => {
+        setShowTooltip(false);
+        if (!selectAll) {
+            setActiveForm(forms);
+        } else {
+            setActiveForm([]);
+        }
+        setSelectAll(!selectAll);
+    };
+
+    /**
+     * The callback used by functions above to set the values of the row.
+     * if clearValue is true, it will use emtpy value, otherwise it will copy the select-all input value
+     */
+    const setValueForAllInputs = (clearValue?: boolean) => {
+        const cm = columnModels[columnModelIndex];
+
+        activeForms?.forEach((formValue: number) => {
+            // ignore the ones that cannot be updated
+            if (appMode === appModes.EDIT && canUpdateValues && !canUpdateValues[`${formValue}-${cm.column.name}`]) {
+                return;
+            }
+            reset(
+                copyOrClearValue(
+                    cm,
+                    getValues(),
+                    foreignKeyData.current,
+                    formValue,
+                    SELECT_ALL_INPUT_FORM_VALUE,
+                    clearValue
+                )
+            );
+        });
+    };
+
+    /* This is to set the width of text area as the width of select-all-row. We have to involve javascript as 
```

Comment:
Use jsdoc style of comment. You should also explain why 1800 and 1200. And also have a note that if we want to change this, we should also change the css rules that you added (you should point to the rule that is using the same numbers).